### DISCUSSION TITLE
Installation error with pip (requirements_versions.txt)

### DISCUSSION THREAD

**Discussion by Riveda72** [lllyasviel/Fooocus#3171] [Q&A] [answered]
I got this error when trying to install the requirements. 
Windows 10 command line, run as administrator (same error if using powershell)
Python 3.12 (win64), installed for all users.

Command:
**pip install -r requirements_versions.txt**

```
  File "<string>", line 204, in get_source_files
        File "C:\Users\Samuel\AppData\Local\Temp\pip-build-env-8secg_pn\overlay\Lib\site-packages\setuptools\_distutils\cmd.py", line 107, in __getattr__
          raise AttributeError(attr)
      AttributeError: cython_sources
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

This is the content in requirements_versions.txt
```
torchsde==0.2.5
einops==0.4.1
transformers==4.30.2
safetensors==0.3.1
accelerate==0.21.0
pyyaml==6.0
Pillow==9.2.0
scipy==1.9.3
tqdm==4.64.1
psutil==5.9.5
pytorch_lightning==1.9.4
omegaconf==2.2.3
gradio==3.41.2
pygit2==1.12.2
opencv-contrib-python==4.8.0.74
httpx==0.24.1
onnxruntime==1.16.3
timm==0.9.2
```

**Comment by Rogala** [comment, marked as answer]
[Python 3.10-11](https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md#subprocess-exited-with-error)

**Reply by Riveda72** [reply]
meaning that it's mandatory to use this version of python? OK :) Thanks, I'll try it
Is there anything elsa that could break my installation?
I mean I was suggested to install python with administrator privileges and to choose for an installation "for all users". is that relevant?

**Comment by mashb1t** [comment]
@Riveda72 Fooocus supports Python 3.10 and 3.11, not yet 3.12.
If you want to use the latest Python version, make sure to manually apply the changes of https://github.com/lllyasviel/Fooocus/pull/2927
